### PR TITLE
on node list, avoid failing on error if there are not staking keys

### PR DESCRIPTION
**Pull request by felipemadero**
Closes https://github.com/ava-labs/avalanche-cli/issues/1678

### CODE REVIEW

File: cmd/nodecmd/list.go [hunk 1]
```diff
@@ -45,29 +45,34 @@ func list(_ *cobra.Command, _ []string) error {
...
-			nodeConfig, err := app.LoadClusterNodeConfig(cloudID)
-			if err != nil {
-				return err
-			}
 			nodeIDStr := "----------------------------------------"
 			if clusterConf.IsAvalancheGoHost(cloudID) {
 				nodeID, err := getNodeID(app.GetNodeInstanceDirPath(cloudID))
 				if err != nil {
-					return err
+					ux.Logger.PrintToUser(logging.Yellow.Wrap("could not obtain node ID for nodes %s: %s"), cloudID, err)
```
Comment (arturrez) [inline cmd/nodecmd/list.go:57]: nit: use RedXToUser instead

File: cmd/nodecmd/list.go [hunk 2]
```diff
@@ -45,29 +45,34 @@ func list(_ *cobra.Command, _ []string) error {
...
 		}
+		nodeIDs := []string{}
 		for _, cloudID := range clusterConf.GetCloudIDs() {
-			nodeConfig, err := app.LoadClusterNodeConfig(cloudID)
-			if err != nil {
-				return err
-			}
 			nodeIDStr := "----------------------------------------"
 			if clusterConf.IsAvalancheGoHost(cloudID) {
 				nodeID, err := getNodeID(app.GetNodeInstanceDirPath(cloudID))
```
Comment (arturrez) [inline cmd/nodecmd/list.go:55]: can be shortened with if nodeID, err:=..., err!=nil; else nodeIDStr = nodeID.String()

File: cmd/nodecmd/list.go [hunk 3]
```diff
@@ -45,29 +45,34 @@ func list(_ *cobra.Command, _ []string) error {
...
 				if err != nil {
-					return err
+					ux.Logger.PrintToUser(logging.Yellow.Wrap("could not obtain node ID for nodes %s: %s"), cloudID, err)
+				} else {
+					nodeIDStr = nodeID.String()
```
Comment (arturrez) [inline cmd/nodecmd/list.go:59]: simpler `nodeIDs = append(nodeIDs, nodeID.String())` here and no need `nodeIDs = append(nodeIDs, nodeIDStr)` later

File: cmd/nodecmd/list.go [hunk 4]
```diff
@@ -45,29 +45,34 @@ func list(_ *cobra.Command, _ []string) error {
...
+		ux.Logger.PrintToUser("Cluster %q (%s)", clusterName, clusterConf.Network.Kind.String())
+		for i, cloudID := range clusterConf.GetCloudIDs() {
+			nodeConfig, err := app.LoadClusterNodeConfig(cloudID)
+			if err != nil {
+				return err
 			}
 			roles := clusterConf.GetHostRoles(nodeConfig)
 			rolesStr := strings.Join(roles, ",")
 			if rolesStr != "" {
 				rolesStr = " [" + rolesStr + "]"
```
Comment (arturrez) [inline cmd/nodecmd/list.go:73]: nit: use fmt.Sprintf

**Review by arturrez — APPROVED** [review]

File: cmd/nodecmd/list.go (hunk 3 above)
Comment (felipemadero) [reply cmd/nodecmd/list.go:59]: it is needed so, for the default case

File: cmd/nodecmd/list.go (hunk 4 above)
Comment (felipemadero) [reply cmd/nodecmd/list.go:73]: I believe it is ok as it

**Pull request merged**